traffic/traffic_info.cpp

```diff
@@ -87,6 +124,241 @@ TrafficInfo TrafficInfo::BuildForTesting(Coloring && coloring)
 
 bool TrafficInfo::ReceiveTrafficData()
 {
+  vector<SpeedGroup> values;
+  if (!ReceiveTrafficValues(values))
+    return false;
+
+  if (m_keys.size() != values.size())
+  {
+    LOG(LWARNING,
+        ("The number of received traffic values does not correspond to the number of keys."));
+    m_availability = Availability::NoData;
```

Comment:
m_coloring is supposed to be guarded by m_availability.
But ok, let's clear it.